ako-crd-operator/internal/utils/utils.go

```diff
@@ -306,3 +313,195 @@ func GetTenantInNamespace(ctx context.Context, client client.Client, namespace s
+	clusterID := configMap.Data["clusterID"]
+	if clusterID == "" {
+		logger.Error("clusterID not found in avi-k8s-config configmap")
+		return fmt.Errorf("clusterID not found in configmap")
+	}
+	clusterName := clusterID
+	clusterIDArr := strings.Split(clusterID, ":")
+	if len(clusterIDArr) > 1 {
+		// Include first 5 characters to add more uniqueness to cluster name
+		clusterName = clusterIDArr[0] + "-" + clusterIDArr[1][:5]
+	} else {
+		if len(clusterID) > 12 {
+			clusterName = clusterID[:12]
+		}
+	}
+
+	// Extract controllerIP from configmap
+	controllerIP := configMap.Data["controllerIP"]
+	if controllerIP == "" {
+		logger.Error("controllerIP not found in avi-k8s-config configmap")
+		return fmt.Errorf("controllerIP not found in configmap")
+	}
+
+	// Set cluster Name, ID and controller IP
+	os.Setenv("CLUSTER_NAME", clusterName)
+	lib.SetClusterID(clusterID)
+	lib.SetControllerIP(controllerIP)
+	logger.Infof("Set clusterName: %s, clusterID: %s, controllerIP: %s", clusterName, clusterID, controllerIP)
+
+	// Validate AVI secret
+	if err := validateAviSecret(ctx, kubeClient, controllerIP, logger); err != nil {
+		logger.Errorf("AVI secret validation failed: %v", err)
+		return err
+	}
+
+	logger.Info("Cluster configuration initialized successfully")
+	return nil
+}
+
+// validateAviSecret validates the AVI secret by creating an AVI client
+func validateAviSecret(ctx context.Context, kubeClient kubernetes.Interface, controllerIP string, logger *utils.AviLogger) error {
+	logger.Info("Validating AVI secret")
+
+	// Get avi-secret
+	aviSecret, err := kubeClient.CoreV1().Secrets(utils.GetAKONamespace()).Get(ctx, lib.AviSecret, metav1.GetOptions{})
+	if err != nil {
+		logger.Errorf("Failed to get avi-secret: %v", err)
+		return err
+	}
+
+	// Extract credentials from secret
+	authToken := string(aviSecret.Data["authtoken"])
+	username := string(aviSecret.Data["username"])
+	password := string(aviSecret.Data["password"])
+	caData := string(aviSecret.Data["certificateAuthorityData"])
+
+	// Validate that required fields are present
+	if username == "" || (password == "" && authToken == "") {
+		return fmt.Errorf("invalid avi-secret: username is required and either password or authtoken must be provided")
+	}
+
+	// Create HTTP transport with certificate
+	transport, isSecure := utils.GetHTTPTransportWithCert(caData)
+
+	// Configure AVI session options
+	options := []func(*session.AviSession) error{
+		session.DisableControllerStatusCheckOnFailure(true),
+		session.SetTransport(transport),
+		session.SetTimeout(120 * time.Second),
+	}
+
+	if !isSecure {
+		options = append(options, session.SetInsecure)
+	}
+
+	// Use authtoken if available, otherwise use password
+	if authToken != "" {
+		options = append(options, session.SetAuthToken(authToken))
+	} else {
+		options = append(options, session.SetPassword(password))
+	}
+
+	// Create AVI client and test connection
+	aviClient, err := clients.NewAviClient(controllerIP, username, options...)
```

Comment:
can we not directly write it like _, err := clients.NewAviClient(...) instead of doing it in line 504?